### DISCUSSION TITLE
How to set file path as h1?

### DISCUSSION THREAD

**Discussion by ooker777** [mkdocs/mkdocs#3217] [Q&A]
# Problem
The titles of my Markdown notes are mostly in the filename, rarely at the `title` key in YAML, and never in the `#` heading. When built, most of the time Mkdocs renders the correct titles and headings, but for notes having the `#` heading right at the start of the note, it takes it as the heading as well.

Say we have the file `folder/filename.md` with the content:
```md
# First heading
Lorem ipsum
# Second heading
Hello world
```

Then in the output file `folder/filename/index.html` we have:
```html
<h1>First heading </h1>
<h2>First heading</h2>
Lorem ipsum
<h2>Second heading</h2>
Hello world
```
Is there a way to tell Mkdocs to have `<h1>filename</h1>` instead, if there is no `title` key in YAML?

# Attemp
I notice in `override\partials\contents.html` there is this code:

```html
<!--
  Hack: check whether the content contains a h1 headline. If it doesn't, the
  page title (or respectively site name) is used as the main headline.
-->
{% if "\x3ch1" not in page.content and not config.extra['no-auto-h1'] %}
<h1>{{ page.title | d(config.site_name, true)}}</h1>
{% endif %}
```

I try removing the if statement, keeping only 
```html
<h1>{{ page.title | d(config.site_name, true)}}</h1>
```
But it doesn't work. I try searching but can't find any solution, just [this conversation](https://github.com/mkdocs/mkdocs/issues/1008 "Page title: filename vs top header · Issue #1008 · mkdocs/mkdocs") on how Mkdocs should handle this by default.

**Comment by Andre601** [comment]
This looks like you are using Materisl for MkDocs?
I ask because this place is reserved for MkDocs stuff and Materisl for MkDocs questions should be posted on their own discussion.

But to still answer your question: Make sure you actually extend the theme in the mkdocs.yml configuration and that you override the right file in the right path.

**Comment by pawamoy** [comment]
Titles in navigation take precedence over H1 Markdown headings I think. Do you set the `nav` in mkdocs.yml?

**Reply by ooker777** [reply]
No I don't

**Comment by oprypin** [comment]
There is currently no way to do this.

If there's no title, MkDocs will always look at the Markdown content.

"content writing" in the meantime has kinda converged on not having multiple H1 tags so maybe you'll have to give up and make these H2.

Or if you're still in, you should be able to use the `hooks` feature to set a page's title by doing the same as this code:
https://github.com/mkdocs/mkdocs/blob/566a5bdda8bc9e66cc07b8f6b3229a967261bebd/mkdocs/structure/pages.py#L255
Later hopefully I'll come back and write exactly how to do this

**Reply by ooker777** [reply]
I've tried copy the `pages.py` file into `overrides/hooks`, then comment out the line 
```python
title = get_markdown_title(self.markdown)
```
It doesn't change

**Reply by ooker777** [reply]
Also, with the "the first `#` is the h1 tag" logic, why does it duplicate the line to the text h2?
```html
<h1>First heading</h1>
<h2>First heading</h2>
```
